Implemented view f

E0011875 opened the pull request:
Please review.

Commit:
1) Replace TaskTabPanel class parameter with logic

Commit:
1) Rename switchTab to switchTabOnCommand TaskTabPanel.java
2) Added switchTabOnClick() to TaskTabPanel.java
3) Rename taskTabPanel to taskTabPane in TaskTabPanel.java/xml to distinguish between a Pane object from a Panel class
4) Added listerner in UiManager to detect manual switching of tab

Commit:
1) Added TYPE_ALL case on switch to demostrate all cases
2) Added author tag to ViewCommandParser

Commit:
1) Added Floating Case in ViewCommand.java and TaskTabPanel
2) Added updateFilteredListToShowFloating() in Model.java (Implementation in ModelManager.java)
3) Added isFloating in ModelManager.java

LiHaoTan commented on src/main/java/seedu/address/ui/MainWindow.java:
```diff
@@ -190,20 +191,32 @@ private void handleExit() {
+        taskTabPanel.switchTabOnCommand(typeOfList);
+    }
+
+    //@author A0135998H
```
Don't need double

LiHaoTan commented on src/main/java/seedu/address/ui/MainWindow.java:
```diff
@@ -190,20 +191,32 @@ private void handleExit() {
+    public void switchTabOnClick() {
+        taskTabPanel.switchTabOnClick();
+    }
+
```
Can add end //@@author here

LiHaoTan commented on src/main/java/seedu/address/ui/MainWindow.java:
```diff
@@ -190,20 +191,32 @@ private void handleExit() {
         raise(new ExitAppRequestEvent());
     }
 
+    //@author A0135998H
```
it's //@@author A0135998H

LiHaoTan commented on src/main/java/seedu/address/ui/TaskTabPanel.java:
```diff
@@ -61,60 +69,93 @@
      * @param placeholder The AnchorPane where the taskTabPanel must be inserted
      */
-    public TaskTabPanel(AnchorPane placeholder, ObservableList<ReadOnlyTask> taskList) {
+    public TaskTabPanel(AnchorPane placeholder, Logic logic) {
```
I personally think you are creating too much coupling if you pass the entire logic in though. So it's up to you if you want to do this.

LiHaoTan commented on src/main/java/seedu/address/model/ModelManager.java:
```diff
@@ -191,4 +200,9 @@ public boolean isOverdue(ReadOnlyTask task) {
 
+  //@@author A0135998H
+    public boolean isFloating(ReadOnlyTask task) {
+        return !(task.getStartEndDateTime()).isPresent() && !(task.getDeadline()).isPresent();
```
The thing about brackets is to help things become more understandable, in this case it looks as though you are negating a `ZonedDateTime` which is not possible so your brackets are wrong.

so it should be `return !(task.getStartEndDateTime().isPresent()) && !(task.getDeadline().isPresent());` if you want to keep the brackets

Commit:
From Codacy:
1) Allow public access to loadTaskPage() & releasesResources() in MainWindow.java
2) Replace == with .equals() in TaskTabPanel.java
From LiHao:
1) Added brackets in ModelManager.java for readability
2) Improve author tag format

LiHaoTan commented on src/main/java/seedu/address/ui/TaskTabPanel.java:
```diff
@@ -140,11 +140,11 @@ public void switchTabOnClick() {
 
             Tab currTaskTab = getTaskTabPane().getSelectionModel().getSelectedItem();
-            if (currTaskTab == allTasksTab) {
+            if (currTaskTab.equals(allTasksTab)) {
```
Seems like u should compare `id` instead?

Commit:
Compare tab.idProperty() instead of tab object.

LiHaoTan commented on src/main/java/seedu/address/ui/TaskTabPanel.java:
```diff
@@ -140,11 +140,11 @@ public void switchTabOnClick() {
 
             Tab currTaskTab = getTaskTabPane().getSelectionModel().getSelectedItem();
-            if (currTaskTab.equals(allTasksTab)) {
+            if (currTaskTab.idProperty() == allTasksTab.idProperty()) {
```
no don't use reference comparison pls. Compare `getId` pls if you want compare `id`

LiHaoTan committed:
Resolve conflicts

LiHaoTan committed:
Merge branch 'master' into view-floating
